Cython/Shadow.py

```diff
@@ -138,9 +138,45 @@ def inline(f, *args, **kwds):
         return f
 
 
-def compile(f):
-    from Cython.Build.Inline import RuntimeCompiledFunction
-    return RuntimeCompiledFunction(f)
+def compile(*args, **kwargs):
+    import re
+    import inspect
+    from functools import wraps, partial
+    from Cython.Build.Inline import cython_inline, IS_PY3
+
+    f_locals = inspect.currentframe().f_back.f_locals
+    _cython_inline = partial(cython_inline, locals=f_locals, **kwargs)
+
+    decorator_pat = re.compile(r'@([.\w]+)[\(]?')
+
+    def decorator(f):
+        if not IS_PY3:
+            f_globals = f.func_globals
+        else:
+            f_globals = f.__globals__
+
+        code = inspect.getsourcelines(f)[0]
+
+        # We need to search for this decorator and strip it from the source
+        m = decorator_pat.match(code[0].strip())
```

Comment:
How do you know the compile decorator is in `code[0]` and not some other line? It seems very reasonable to add other decorators on top. Those should then actually be ignored in the compilation and only lower decorators should be included. The later decorators should then automatically be added by Python. Just try a simple pass-through decorator on top and below that `print()`s something into the doctest output when applied.

Reply:
Yeah, I haven't addressed/resolved [your earlier comment](https://github.com/cython/cython/pull/3671#discussion_r438645738), yet.  I've only fixed some of the small things you mentioned last time in this last rebase.

Otherwise, I have some questions that I'll post in that comment.